core/templates/dev/head/filters.js

```diff
@@ -240,6 +240,15 @@ oppia.filter('parameterizeRuleDescription', [
           replacementText = inputs[varName];
         }
 
+        if (typeof (replacementText) === 'string') {
+          // Replacing all occurances of $ with $$.
+          // When this is used in the next replace
+          // statement, it will be converted back to $.
+          // Also to replace by $$, the appropriate string is $$$
```

Comment:
I feel like this is very error prone. What about splitting the string by $ and processing each component separately?

Reply:
Actually while searching I found that the split, replace & then join and the regex do the same exact thing but the regex is better at performance. Lost the link though.. Will reference it here if I find it again. But if you suggest to go ahead with a split and recombine strategy I could implement that instead. Thanks

Reply:
Changed this to use split.